legend-depot-store-metrics/src/main/java/org/finos/legend/depot/store/metrics/services/QueryMetricsHandler.java

```diff
@@ -70,6 +73,14 @@ public List<VersionQueryMetric> findSnapshotVersionMetricsBefore(Date date)
         return metricsStore.findMetricsBefore(date).parallelStream().filter(metric -> VersionValidator.isSnapshotVersion(metric.getVersionId())).collect(Collectors.toList());
     }
 
+    public void persistMetrics()
+    {
+        while (queryMetricsRegistry.findFirst() != null)
```

Comment:
how is this going to work in a concurrent enviorment where the queue is being modified while this is happening

Reply:
the underlying implementation is a concurrent queue, and it uses the FIFO rather than an iterator which needs a locking mech, so even if the queue is being modified, it depends which operation completed first the polling or the recording which would decide whether this comes out to be true or not